pkg/config/dynamicconfig.go

```diff
@@ -0,0 +1,25 @@
+package config
```

Comment:
Hmm, are we creating enough new types here that it makes sense to create a new package for dynamic config? `pkg/config/dynconfig` or something?

Reply:
Though its only 3 new types: Loader, LoaderConfig, Datasource